js_modules/dagster-ui/packages/ui-core/src/sensors/SensorTargetList.tsx

```diff
@@ -35,3 +64,131 @@ export const SensorTargetList = ({
+      .slice()
+      .sort((a, b) =>
+        COMMON_COLLATOR.compare(displayNameForAssetKey(a.key), displayNameForAssetKey(b.key)),
+      );
+  }, [assetSelection.assets]);
+
+  const assetsWithAMP = React.useMemo(
+    () => sortedAssets.filter((asset) => !!asset.definition?.autoMaterializePolicy),
+    [sortedAssets],
+  );
+  const assetsWithoutAMP = React.useMemo(
+    () => sortedAssets.filter((asset) => !asset.definition?.autoMaterializePolicy),
+    [sortedAssets],
+  );
+
+  return (
+    <>
+      <Dialog
+        isOpen={showAssetSelection}
+        title="Targeted assets"
+        onClose={() => setShowAssetSelection(false)}
+        style={{width: '750px', maxWidth: '80vw', minWidth: '500px', transform: 'scale(1)'}}
+        canOutsideClickClose
+        canEscapeKeyClose
+      >
+        <Box flex={{direction: 'column'}}>
+          {assetsWithAMP.length ? (
+            <Section
+              title="Assets with a materialization policy"
+              titleBorder="bottom"
+              assets={assetsWithAMP}
+            />
+          ) : null}
+          {assetsWithoutAMP.length ? (
+            <Section
+              title="Assets without a materialization policy"
+              titleBorder="top-and-bottom"
+              assets={assetsWithoutAMP}
+            />
+          ) : null}
+        </Box>
+        <DialogFooter topBorder>
+          <Button
+            intent="primary"
+            onClick={() => {
+              setShowAssetSelection(false);
+            }}
+          >
+            Close
+          </Button>
+        </DialogFooter>
+      </Dialog>
+      <ButtonLink
+        onClick={() => {
+          setShowAssetSelection(true);
+        }}
+      >
+        {assetSelection.assetSelectionString}
+      </ButtonLink>
+    </>
+  );
+};
+
+const Section = ({
+  assets,
+  title,
+  titleBorder = 'top-and-bottom',
+}: {
+  assets: SensorAssetSelectionFragment['assets'];
+  title: string;
+  titleBorder: React.ComponentProps<typeof Box>['border'];
+}) => {
+  const [isOpen, setIsOpen] = React.useState(true);
+  return (
+    <>
+      <Box border={titleBorder} padding={{right: 24, vertical: 12}}>
+        <Box
+          flex={{direction: 'row', gap: 4}}
+          style={{cursor: 'pointer'}}
+          onClick={() => {
+            setIsOpen(!isOpen);
+          }}
+        >
+          <ToggleButton onToggle={() => {}} isOpen={isOpen} />
+          <Subtitle2>
+            {title} ({numberFormatter.format(assets.length)})
+          </Subtitle2>
+        </Box>
+      </Box>
+      {isOpen ? (
+        <div style={{maxHeight: '300px', overflowY: 'scroll'}}>
+          <VirtualizedItemListForDialog
+            padding={0}
+            items={assets}
+            renderItem={(asset) => <VirtualizedSelectedAssetRow asset={asset} key={asset.id} />}
+            itemBorders
+          />
+        </div>
+      ) : null}
+    </>
+  );
+};
+
+const VirtualizedSelectedAssetRow = ({
```

Comment:
I wonder if this could be merged with `renderItemAssetKey` in the AssetTagCollections dialog? Probably not worth it but we do have a bunch of dialogs that show lists of asset keys now